### DISCUSSION TITLE
ignore stuff starting with period

### DISCUSSION THREAD

**Discussion by joaociocca** [epi052/feroxbuster#693] [Q&A] [answered]
How can I make ferox not scan things starting with `.`? I thought a simple `--dont-scan '^\..*'` should do the trick, but it complains that it could not determine initial targets because the regex matches the URL, so the scan will never start.
I ask this to prevent ferox, on more thorough modes (but not exclusively `--thorough`) to try and scan for `<url>/.htpasswds/<anything, actually>` because it seem to dive in a rabbit hole from there.

An example, I was trying to show VulnHub's Jangow to a friend, and it spiralled down trying to search for things like...
![image](https://user-images.githubusercontent.com/6473725/198341665-799670fe-018f-4cb4-94cb-a1b9f7d0b0f5.png)

**Comment by epi052** [comment]
Howdy! You probably could add the base url to the regex and be ok.

**Reply by joaociocca** [reply]
looks like the implementation of `--dont-scan` changes `\.` into `/.` instead of escaping the `.`
![image](https://user-images.githubusercontent.com/6473725/198342761-25f0b441-aade-4d17-8884-8979e527546d.png)

**Reply by epi052** [reply]
Ok, that was off the cuff. I'll take a look this evening

**Reply by epi052** [reply]
Actually, try the base url plus a period, no escape, no second period/asterisk

**Reply by joaociocca** [reply]
Nope.
![image](https://user-images.githubusercontent.com/6473725/198351764-ffeedc2c-bd13-4b02-be85-430e65c2698e.png)

I also tried using character class/set `[.]` and it also gets bypassed
![image](https://user-images.githubusercontent.com/6473725/198351932-149bf195-df2b-44cc-b6b2-0cc4de5969df.png)

**Comment by epi052** [comment]
👍 I'll need to actually look at the code it seems 😂

**Reply by joaociocca** [reply]
"it's regex, but not quite" =p times like this I wish I wasn't so blocked when trying to learn programming to give a hand. But I can help with test, if needed! I ended up on this scenario trying to see if I can find with ferox a link I've found manually on Jangow. I couldn't find it with other tools like dirb/dirbuster/gobuster.

**Comment by epi052** [comment, marked as answer]
Ok, this should behave the way you're wanting

```
--dont-scan '/\..*' 
```

`--dont-scan` does double duty. When it's an absolute/valid url that `url::Url::parse` can successfully ingest, then the comparison is done as a `Url` -> `Url`. On the other hand, if `url::Url::parse` fails, we parse what's given to `dont-scan` as a fullblown regex. So, when we were trying things with http... we weren't hitting the regex branch.

**Reply by joaociocca** [reply]
DING DING DING! Awesome, epi! Thank you! <3
it hit the right target I was aiming to, without polluting the output with all that .ht thrash! =D

![image](https://user-images.githubusercontent.com/6473725/198494627-58c2f6ae-b558-4bd5-b61c-a2022cd390e9.png)
